File: app/src/main/java/me/instabattle/app/activities/PhotoEditActivity.java
package me.instabattle.app.activities;

import android.app.Activity;
import android.content.Intent;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import me.instabattle.app.R;
import me.instabattle.app.images.Util;
import me.instabattle.app.managers.PhotoManager;
import me.instabattle.app.models.Entry;
import me.instabattle.app.settings.State;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class PhotoEditActivity extends Activity {

    private static final String TAG = "PhotoEditActivity";

    public static byte[] photoBytes;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_photo_edit);

        String battleTitle = getIntent().getStringExtra("battleTitle");

        ((TextView) findViewById(R.id.editPhotoBattleTitle)).setText(battleTitle);

        Bitmap photoBitmap = Util.decodeSampledBitmapFromBytes(photoBytes, 256, 256);
        ((ImageView) findViewById(R.id.currentPhoto)).setImageBitmap(photoBitmap);

        Log.d(TAG, "Called onCreate");
    }

    public void takeNewPhoto(View v) {
        Intent camera = new Intent(this, CameraActivity.class);
        startActivity(camera);
    }

    public void useThisPhoto(View v) {
        if (!State.creatingBattle) {
            State.chosenBattle.createEntryAndDo(new Callback<Entry>() {
                @Override
                public void onResponse(Call<Entry> call, Response<Entry> response) {
                    PhotoManager.upload(response.body().getImageName(), photoBytes);
                }

                @Override
                public void onFailure(Call<Entry> call, Throwable t) {
                    //TODO
                    Log.e(TAG, "failed to add new entry: " + t);
                }
            });
            Intent goToBattle = new Intent(this, BattleActivity.class);
            startActivity(goToBattle);
        } else {
            Intent goToNewBattle = new Intent(this, CreateBattleActivity.class);
            CreateBattleActivity.photoBytes = photoBytes;
            startActivity(goToNewBattle);
        }
    }
}

File: app/src/main/kotlin/me/instabattle/app/activities/PhotoEditActivity.kt
package me.instabattle.app.activities

import android.os.Bundle
import android.view.View
import kotlinx.android.synthetic.main.activity_photo_edit.*
import me.instabattle.app.R
import me.instabattle.app.images.Util
import me.instabattle.app.managers.PhotoManager
import me.instabattle.app.models.Entry
import me.instabattle.app.settings.GlobalState
import org.jetbrains.anko.error
import org.jetbrains.anko.startActivity
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class PhotoEditActivity : DefaultActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_photo_edit)

        editPhotoBattleTitle.text = intent.getStringExtra("battleTitle")

        val photoBitmap = Util.decodeSampledBitmapFromBytes(photoBytes!!, 256, 256)
        currentPhoto.setImageBitmap(photoBitmap)
    }

    fun takeNewPhoto(v: View) = startActivity<CameraViewActivity>()

    fun useThisPhoto(v: View) {
        if (!GlobalState.creatingBattle) {
            GlobalState.chosenBattle!!.createEntryAndDo(object : Callback<Entry> {
                override fun onResponse(call: Call<Entry>, response: Response<Entry>) {
                    PhotoManager.upload(response.body()!!.imageName!!, photoBytes!!)
                }

                override fun onFailure(call: Call<Entry>, t: Throwable) {
                    //TODO
                    error {"failed to add new entry: $t"}
                }
            })
            startActivity<BattleActivity>()
        } else {
            //CreateBattleActivity.photoBytes = photoBytes
            startActivity<CreateBattleActivity>()
        }
    }

    companion object {
        var photoBytes: ByteArray? = null
    }
}
